Model/Indexer/ProductIndexer.php

```diff
@@ -128,6 +166,58 @@ public function doIndex(Store $store, array $ids = [])
             $collection,
             $store
         );
+        $this->handleDeletedProducts($collection, $store, $ids);
+    }
+
+    /**
+     * @param ProductCollection $existingCollection
+     * @param Store $store
+     * @param array $givenIds
+     * @throws NostoException
+     */
+    private function handleDeletedProducts(ProductCollection $existingCollection, Store $store, array $givenIds)
+    {
+        if (!empty($givenIds)) {
+
+            if ($this->isSchedule()) {
+                $searchCriteria = $this->searchCriteriaBuilder
+                    ->addFilter('action', ProductIndexerExcludeInterface::ACTION_DELETE, 'eq')
+                    ->addFilter('entity_id', $givenIds, 'in')
+                    ->create();
+
+                /** @var ExcludeRepository $indexerExcludeRepository */
+                $indexerExcludeRepository = $this->excludeRepositoryFactory->create();
+
+                $items = $indexerExcludeRepository->search($searchCriteria)->getItems();
+
+                $ignoreIds = array_map(function ($item) {
+                    return $item->getEntityId();
+                }, $items);
+
+                $givenIds = array_diff($givenIds, $ignoreIds);
+                $indexerExcludeRepository->deleteByIds($ignoreIds);
```

Comment:
Can be extracted to a separate method. If I understood this correctly, you're removing the given ID's from the exclude table. In which situation would I add twice to that exclude table?

Reply:
Right now I don think any situation you can add twice into table but at least I don want table bigger